react/week1/todo-app/src/App.js

```diff
@@ -0,0 +1,81 @@
+import logo from './logo.svg';
+import './App.css';
+
+function DateItem(props) {
+  let d = new Date();
+  d.setFullYear(props.date.year, props.date.month - 1, props.date.day);
+  const today = new Date();
+  const validDate = (d > today) ? 'valid': 'expired';
+  return (
+    <p>{props.date.day}/ {props.date.month}/ {props.date.year} {`${validDate}`}</p>
+  )
+}
+
+function TodoItem(props) {
+  return (
+    <li>
+      <h2>#{props.number}: {props.name}</h2>
+      <DateItem date={props.deadlineDate}></DateItem>
+    </li>
+  )
+}
+
+function TodoList(props) {
+  return (
+    <ul>
+      <h1>Todo List</h1>
+      {
+        props.todos.map(todo => {
```

Comment:
If you use destructuring then instead of `props.todos.map` you can simply do `todos.map`